### PR TITLE
Person picture API additions for bug fix

### PR DESCRIPTION
**Pull request by jevansaks**
Fixing a product bug (issue Microsoft/microsoft-ui-xaml#311, PR Microsoft/microsoft-ui-xaml#358) requires adding a TemplateSettings API. 

Hopefully this doc is enough for the process, but let me know if there's more to do (like putting the other repo's issue into the API docs project board, etc).

### CODE REVIEW

**Comment by jesbis** [comment]
> let me know if there's more to do (like putting the other repo's issue into the API docs project board, etc).

Yeah we should track API changes on the API board - I added the main repo issue to the [API review status board](https://github.com/Microsoft/microsoft-ui-xaml/projects/5). 

Please also:
- add a link to this doc from the main issue
- move the main issue to the "ready for API review" column when ready for API review and ping MikeHillberg

File: active/personpicture/PersonPictureAPI.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+The best practice for controls is that the XAML file includes all UI related colors, sizes and visual states.
+XAML markup has expressiveness limitations that require the controls to do some calculations or value conversions
+in code-behind and expose those results to the XAML to bind from. The pattern we have established for that is 
+having a TemplateSettings property on the control of type <I>Foo</I>Control, where <I>Foo</I> is the name of 
+the control. See [this doc page](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/template-settings-classes)
+for more details.
+
+# Description
+
+PersonPicture.TemplateSettings and the new type PersonPictureTemplateSettings are needed to help move some
```
Comment (jesbis) [inline active/personpicture/PersonPictureAPI.md]: We're trying to use specs as a draft for what will go on docs.microsoft.com.

I think this is background info that could go in the Background section (which doesn't go to DMC), and the description (which goes to DMC) could be something like:

> Provides calculated values that can be referenced as TemplatedParent sources when defining templates for a PersonPicture control. Not intended for general use.

File: active/personpicture/PersonPictureAPI.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        PersonPictureTemplateSettings TemplateSettings { get; };
+
+        ...
+    }
+
+    [webhosthidden]
+    runtimeclass PersonPictureTemplateSettings : Windows.UI.Xaml.DependencyObject
+    {
+        String EffectiveInitials { get; };
```
Comment (jesbis) [inline active/personpicture/PersonPictureAPI.md]: Could you fill out the API Notes spec section for these props with a 1 or 2 line description of what they do? That's what'll get published to docs.microsoft.com. See other PRs for examples.

File: active/personpicture/PersonPictureAPI.md (hunk 1 above)
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: Sounds good, took that suggestion.

File: active/personpicture/PersonPictureAPI.md (hunk 2 above)
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: I added an API Notes section.

**Comment by jevansaks** [comment]
I replied to PR feedback and linked this PR from the main issue. Here's the ping for @MikeHillberg to come take a look at this as well. :)

File: active/personpicture/PersonPictureAPI.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+visual state management logic out of the PersonPicture code implementation into the XAML file.
+
+
+# Description
+
+Provides calculated values that can be referenced as TemplatedParent sources when defining templates for a PersonPicture control. Not intended for general use.
+
+# API Details
+
+### PersonPicture
```
Comment (MikeHillberg) [inline active/personpicture/PersonPictureAPI.md]: Why the level 3 heading under a level 1?

File: active/personpicture/PersonPictureAPI.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        String EffectiveInitials { get; };
+        Windows.UI.Xaml.Media.ImageBrush EffectiveImageBrush { get; };
+    }
+
+}
+```
+
+# API Notes
+
+| Name | Description |
```
Comment (MikeHillberg) [inline active/personpicture/PersonPictureAPI.md]: Could you add the class name above this table, and make the first column "Member Name"?  So the structure becomes [class, member table], [class, member table], ...

File: active/personpicture/PersonPictureAPI.md (hunk 3 above)
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: I was following the other specs, but I think you left that comment in another too. :)

File: active/personpicture/PersonPictureAPI.md (hunk 4 above)
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: Ok.

File: active/personpicture/PersonPictureAPI.md (hunk 3 above)
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: I've removed the heading actually.

File: active/personpicture/PersonPictureAPI.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+for more details. 
+
+PersonPicture.TemplateSettings and the new type PersonPictureTemplateSettings are needed to help move some
+visual state management logic out of the PersonPicture code implementation into the XAML file.
+
+
+# Description
+
+Provides calculated values that can be referenced as TemplatedParent sources when defining templates for a PersonPicture control. Not intended for general use.
+
```
[the remark this answers is not in the document]
Comment (jevansaks) [reply active/personpicture/PersonPictureAPI.md]: None of our templatesettings types have an example, nor do I think they should. They're not for general use. https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.comboboxtemplatesettings

[the remark this answers is not in the document]
Comment (MikeHillberg) [reply active/personpicture/PersonPictureAPI.md]: Our current documentation is not always representative of where we want to be, and an example says a thousand words.  It doesn't need to be complicated, e.g. borrowing from AppBarButton's KeyboardAcceleratorTextMinWidth template setting property spec:

```Xaml
<ControlTemplate TargetType="AppBarButton"> 
        <Grid x:Name="Root" 
            MinWidth="{TemplateBinding MinWidth}"  … > 
            … 
            <TextBlock x:Name="KeyboardAcceleratorTextLabel" 
                MinWidth="{Binding RelativeSource={RelativeSource TemplatedParent},  
                    Path=TemplateSettings.KeyboardAcceleratorTextMinWidth}" 
                /> 
            … 
 
```Xaml

**Pull request merged**